ST_PolygonFromEnvelope bug

test code:
```python
def test_ST_PolygonFromEnvelope():
    x_min = pandas.Series([0.0 for x in range(1, 40000001)])
    x_max = pandas.Series([1.0 for x in range(1, 40000001)])
    y_min = pandas.Series([2.0 for x in range(1, 40000001)])
    y_max = pandas.Series([3.0 for x in range(1, 40000001)])

    rst = arctern.ST_PolygonFromEnvelope(x_min, y_min, x_max, y_max)
    assert len(rst) == 40000000
```
test result:
```
 AssertionError: assert 5363167 == 40000000
```

--- a/cpp/src/gis/gdal/gis_functions.cpp
+++ b/cpp/src/gis/gdal/gis_functions.cpp
@@ -472,6 +472,7 @@ std::vector<std::shared_ptr<arrow::Array>> ST_PolygonFromEnvelope(
         OGRPolygon polygon;
         polygon.addRing(&ring);
         auto array_ptr = AppendWkb(builder, &polygon);
+        if (array_ptr != nullptr) result_array.push_back(array_ptr);
       }
     }
   }

--- a/python/tests/geo/geo_test.py
+++ b/python/tests/geo/geo_test.py
@@ -555,6 +555,14 @@ def test_ST_Buffer():
 
     assert rst[0] == expect
 
+def test_ST_PolygonFromEnvelope2():
+    x_min = pandas.Series([0.0 for x in range(1, 40000001)])
+    x_max = pandas.Series([1.0 for x in range(1, 40000001)])
+    y_min = pandas.Series([2.0 for x in range(1, 40000001)])
+    y_max = pandas.Series([3.0 for x in range(1, 40000001)])
+
+    rst = arctern.ST_PolygonFromEnvelope(x_min, y_min, x_max, y_max)
+    assert len(rst) == 40000000
 
 def test_ST_PolygonFromEnvelope():
     x_min = pandas.Series([0.0])

